packages/lesswrong/components/ultraFeed/UltraFeedCommentsDialog.tsx

```diff
@@ -133,17 +137,27 @@ const UltraFeedCommentsDialog = ({
   // scroll to comment clicked on when dialog opens
   useEffect(() => {
     let scrollTimer: NodeJS.Timeout | null = null;
+    let fadeTimer: NodeJS.Timeout | null = null;
 
     if (!isLoading && targetCommentId && comments && comments.length > 0) {
       scrollTimer = setTimeout(() => {
         const element = window.document.getElementById(targetCommentId);
 
-        const container = element?.closest('.MuiDialogContent-root') as HTMLElement | null;
-
-        if (element && container) {
-          const isScrollable = container.scrollHeight > container.clientHeight;
-
-          if (isScrollable) {
+        if (element) {
+          const container = (() => {
+            let node: HTMLElement | null = element.parentElement;
+            while (node) {
+              const style = window.getComputedStyle(node);
+              const overflowY = style.overflowY;
+              if ((overflowY === 'auto' || overflowY === 'scroll' || overflowY === 'overlay') && node.scrollHeight > node.clientHeight) {
+                return node;
+              }
+              node = node.parentElement;
+            }
+            return null;
+          })();
```

Comment:
This needs a comment explaining what it's doing, also extract the IIFE out to an actual function (ideally not defined inside the component)